charts/consul/templates/client-dns-clusterrole.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if (or (and (ne (.Values.client.enabled | toString) "-") .Values.client.enabled) (and (eq (.Values.client.enabled | toString) "-") .Values.global.enabled)) }}
```

Comment:
This is not currently necessary, this would only be necessary if we decided to use the kube api to get the kubedns service ip. Left it here in case we decide to go that route. The API request is also in a prev commit in this PR.